### ISSUE TITLE
Multi-thread compression

### ISSUE THREAD

**Issue by TalgatAkhm** [srikanth-lingala/zip4j#319]
Hi! 
Thanks for the opportunities in zip file handling provided in the library
I've read the discussion in #229 and want to give my feedback about parallel compression and zip file building. 

I've tested parallel compression using [Apache Common Compress](https://github.com/apache/commons-compress) library and single thread compression using zip4j. 

Input data: 19 .exe files of ~200mb
CPU: Intel core i7-8700K
SSD: Samsung 970 evo plus

Time for Deflate compression in the same compression level:
| Info | Time |
| ----- | ----- |
| Apache Common Compress, parallel zip creation | ~20sec |
| Zip4j, ZipFile | ~90sec |

Example of parallel compression code usage you can find in [ACC tests](https://github.com/apache/commons-compress/blob/02dbe7bb52d541271c3048728c156ca2f1034bc0/src/test/java/org/apache/commons/compress/archivers/zip/ParallelScatterZipCreatorTest.java)

It seems to me, that the time difference allows to say that the parallel compression functionality is useful. 
Is it possible to implement parallel archiving in Zip4j? 

P.S.
As for sequential writing to zip file and impossibility for parallel file writing: As far as I can understand, writing to zip file in ACC is sequential and synchronized, but compression is parallel. I mean ```DeflaterOutputStream``` or ```Deflater``` is used multithreaded - the content was saved to temp files or into streams (I don't actually precisely know it), and then all of this entries and their compressed content were merged into the zip file in one thread.

**Comment by srikanth-lingala** [comment]
Hi. Thanks for your detailed analysis. I will have a look into the sample code and ACC, and post an update here.

**Comment by RinOrz** [comment]
@srikanth-lingala Excuse me, I want to know if there is any progress?

**Comment by srikanth-lingala** [comment]
@RinOrz Unfortunately there is no progress yet. I am extremely short on free time now a days, and I did not find time to work on this feature yet.

**Comment by lzxshare** [comment]
> @RinOrz不幸的是，目前还没有进展。我现在一天的空闲时间非常短，而且我还没有找到时间来处理这个功能。
yes,i think Multi-thread is very useful in  some scenarios，especially when dealing with large files,  I hope the author can realize this function as soon as possible，many thanks
